src/Base/UnitsApi.cpp

```diff
@@ -139,23 +127,35 @@ void UnitsApi::setSchema(UnitSystem system)
                                        // Quantity (e.g. mi=1.8km rather then 1.6km).
 }
 
-QString UnitsApi::toString(const Base::Quantity& quantity, const QuantityFormat& format)
+std::string UnitsApi::toString(const Base::Quantity& quantity, const QuantityFormat& format)
 {
-    QString value = QString::fromLatin1("'%1 %2'")
-                        .arg(quantity.getValue(), 0, format.toFormat(), format.precision)
-                        .arg(quantity.getUnit().getString());
-    return value;
+    std::stringstream ss;
+    ss << '\'' << toNumber(quantity, format) << ' ' << quantity.getUnit().getString() << '\'';
+    return ss.str();

```

Comment:
```suggestion
    return fmt::format("'{} {}'", toNumber(quantity, format), quantity.getUnit().getString());
```

Maybe it wopuld be possible to do that without `toNumber()` method with something like `{:{}{}}` as first arg, but this does not make it more readable so I don't think that it would be a good change.